src/Data/User/UserDataClass.ts

```diff
@@ -1,8 +1,25 @@
 import { provideLocalStorageDataClass } from '../../utils/provideLocalStorageDataClass'
 import richterImage from './FakeBinaries/richter.jpg'
 import braschauImage from './FakeBinaries/braschau.jpg'
+import { CurrentUser } from '../CurrentUser/CurrentUserDataItem'
+import { load } from 'scrivito'
 
 export const User = provideLocalStorageDataClass('User', {
+  postProcessData: async (data) => {
+    const pisaUserId = await load(() => CurrentUser.get('pisaUserId'))
+    if (data._id !== pisaUserId) return data
```

Comment:
* [ ] FYI Only processing the user without also processing the list of users looks like there's a piece missing. Shouldn't we also inject the user to the index? Maybe an explanatory comment is missing?

Reply:
`postProcessData` is used for all `index`, `get` and `update` functions. See https://github.com/Scrivito/scrivito-portal-app/pull/305

Reply:
Yes, but I read the code above as if it post processes a GET for user with a certain ID, without post processing the INDEX to make sure the user is present. If my observation is correct, my question is: how does the system learn about the user?

Reply:
> If my observation is correct, my question is: how does the system learn about the user?

Pisa knows about the current user - otherwise you would not be able to communicate with Pisa. So Pisas `whoami` endpoint will provide you with a pisa user ID. The `user` endpoint of pisa understands this user ID and returns the data for that user.

Reply:
But Pisa will take care of this. No need to inject it manually.

Reply:
I still don't get it. The users are provided using `provideLocalStorageDataClass`. Pisa is not involved in generating the list of users.

Reply:
`provideLocalStorageDataClass('User'` is only meant as a stand-in for a real pisa instance, so we can develop fairly independent of an actual instance running.

The fake user data has this:

https://github.com/Scrivito/scrivito-portal-app/blob/effb95937f70c97975e626fc2d6aaa2e75e382ef/src/Data/User/UserDataClass.ts#L71-L78

And the partially fake current user info has this:

https://github.com/Scrivito/scrivito-portal-app/blob/effb95937f70c97975e626fc2d6aaa2e75e382ef/src/Data/CurrentUser/CurrentUserDataItem.ts#L23

With that "Markus Neumann" becomes the currently logged in user with the information provided by IAM and Neoletter.

Reply:
Thank! The hardcoded `F87BDC400E41D630E030A8C00D01158A` was the missing link. I had actually eyeballed the fake users expecting to find the ID there but overlooked it. A comment (or a constant) would have helped.